### ISSUE TITLE
[BUG] Unable to import from paperless-ng

### ISSUE THREAD

**Issue by tht** [paperless-ngx/paperless-ngx#282]
**Describe the bug**
I'm a paperless-ng user trying to migrate to paperless-ngx. I'm using the docker version and fail horribly no matter how I try.

**Trying in-place migration as described in documentation**
First I followed the recommended route and only changed the docker image to paperless-ngx. When starting up I got these errors on the console:

```
ebserver_1  | Adjusting permissions of paperless files. This may take a while.
webserver_1  | Apply database migrations...
webserver_1  | /usr/local/lib/python3.9/site-packages/dateparser/date_parser.py:35: PytzUsageWarning: The localize method is no longer necessary, as this time zone supports the fold attribute (PEP 495). For more details on migrating to a PEP 495-compliant implementation, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
webserver_1  |   date_obj = stz.localize(date_obj)
webserver_1  | Operations to perform:
webserver_1  |   Apply all migrations: admin, auth, authtoken, contenttypes, django_q, documents, paperless_mail, sessions
webserver_1  | Running migrations:
webserver_1  |   Applying django_q.0014_schedule_cluster... OK
webserver_1  |   Applying documents.1012_fix_archive_files...Traceback (most recent call last):
webserver_1  |   File "/usr/src/paperless/src/manage.py", line 11, in <module>
webserver_1  |     execute_from_command_line(sys.argv)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
webserver_1  |     utility.execute()
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
webserver_1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
webserver_1  |     self.execute(*args, **cmd_options)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
webserver_1  |     output = self.handle(*args, **options)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 89, in wrapped
webserver_1  |     res = handle_func(*args, **kwargs)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 244, in handle
webserver_1  |     post_migrate_state = executor.migrate(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 117, in migrate
webserver_1  |     state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
webserver_1  |     state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
webserver_1  |     state = migration.apply(state, schema_editor)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/migrations/migration.py", line 126, in apply
webserver_1  |     operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
webserver_1  |     self.code(from_state.apps, schema_editor)
webserver_1  |   File "/usr/src/paperless/src/documents/migrations/1012_fix_archive_files.py", line 235, in move_old_to_new_locations
webserver_1  |     raise ValueError(
webserver_1  | ValueError: Archived document ID:428 does not exist at: /usr/src/paperless/src/../media/documents/archive/2017/none/2017_10_25 IBA Rechnung Akonto Q3_2017.pdf
webserver_1  | Search index out of date. Updating...
webserver_1  | /usr/local/lib/python3.9/site-packages/dateparser/date_parser.py:35: PytzUsageWarning: The localize method is no longer necessary, as this time zone supports the fold attribute (PEP 495). For more details on migrating to a PEP 495-compliant implementation, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
webserver_1  |   date_obj = stz.localize(date_obj)
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
webserver_1  |     return self.cursor.execute(sql, params)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
webserver_1  |     return Database.Cursor.execute(self, query, params)
webserver_1  | sqlite3.OperationalError: no such column: documents_document.archive_filename
webserver_1  |
webserver_1  | The above exception was the direct cause of the following exception:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/src/paperless/src/manage.py", line 11, in <module>
webserver_1  |     execute_from_command_line(sys.argv)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
webserver_1  |     utility.execute()
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
webserver_1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
webserver_1  |     self.execute(*args, **cmd_options)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
webserver_1  |     output = self.handle(*args, **options)
webserver_1  |   File "/usr/src/paperless/src/documents/management/commands/document_index.py", line 23, in handle
webserver_1  |     index_reindex(progress_bar_disable=options["no_progress_bar"])
webserver_1  |   File "/usr/src/paperless/src/documents/tasks.py", line 29, in index_reindex
webserver_1  |     for document in tqdm.tqdm(documents, disable=progress_bar_disable):
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/tqdm/std.py", line 989, in __init__
webserver_1  |     total = len(iterable)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 262, in __len__
webserver_1  |     self._fetch_all()
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 1324, in _fetch_all
webserver_1  |     self._result_cache = list(self._iterable_class(self))
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 51, in __iter__
webserver_1  |     results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
webserver_1  |     cursor.execute(sql, params)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 66, in execute
webserver_1  |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
webserver_1  |     return executor(sql, params, many, context)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
webserver_1  |     return self.cursor.execute(sql, params)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
webserver_1  |     raise dj_exc_value.with_traceback(traceback) from exc_value
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
webserver_1  |     return self.cursor.execute(sql, params)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
webserver_1  |     return Database.Cursor.execute(self, query, params)
webserver_1  | django.db.utils.OperationalError: no such column: documents_document.archive_filename
webserver_1  | Exception ignored in: <function tqdm.__del__ at 0x7f3901934f70>
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/tqdm/std.py", line 1147, in __del__
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/tqdm/std.py", line 1266, in close
webserver_1  | AttributeError: 'tqdm' object has no attribute 'disable'
```

It still managed to somehow start up but started to throw exceptions without showing any files in the web ui.

```
django.db.utils.OperationalError: no such column: documents_document.archive_filename
```

**Trying using `document_exporter`/`document_importer`**
I gave up on this method to migrate and restored paperless-ng from a backup. All was running fine in paperless-ng when I used `document_exporter` in paperless-ng to export all files to a directory.

After the export I removed all the files from the persistent volumes (except for the export of course) and started a fresh installation of paperless-ngx. Added the superuser and logged in. All looked good until I tried to import the export from paperless-ng.

```
# docker exec -it paperless_webserver_1 python manage.py document_importer /usr/src/paperless/data/export
/usr/local/lib/python3.9/site-packages/dateparser/date_parser.py:35: PytzUsageWarning: The localize method is no longer necessary, as this time zone supports the fold attribute (PEP 495). For more details on migrating to a PEP 495-compliant implementation, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
  date_obj = stz.localize(date_obj)
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/models/options.py", line 608, in get_field
    return self.fields_map[field_name]
KeyError: 'colour'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/serializers/json.py", line 70, in Deserializer
    yield from PythonDeserializer(objects, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/serializers/python.py", line 119, in Deserializer
    field = Model._meta.get_field(field_name)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/options.py", line 610, in get_field
    raise FieldDoesNotExist("%s has no field named '%s'" % (self.object_name, field_name))
django.core.exceptions.FieldDoesNotExist: Tag has no field named 'colour'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/paperless/src/manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/src/paperless/src/documents/management/commands/document_importer.py", line 84, in handle
    call_command("loaddata", manifest_path)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 181, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 78, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 123, in loaddata
    self.load_label(fixture_label)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 181, in load_label
    for obj in objects:
  File "/usr/local/lib/python3.9/site-packages/django/core/serializers/json.py", line 74, in Deserializer
    raise DeserializationError() from exc
django.core.serializers.base.DeserializationError: Problem installing fixture '/usr/src/paperless/data/export/manifest.json':
```

Ok, something wrong with the key `colour`. I manually removed all `colour` attributes from all the tags in the manifest file and tried again. Next error:

```
/usr/local/lib/python3.9/site-packages/dateparser/date_parser.py:35: PytzUsageWarning: The localize method is no longer necessary, as this time zone supports the fold attribute (PEP 495). For more details on migrating to a PEP 495-compliant implementation, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
  date_obj = stz.localize(date_obj)
Installed 1507 object(s) from 1 fixture(s)
Copy files into paperless...
  0%|                                                                                                                                                                                             | 0/1404 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/usr/src/paperless/src/manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/src/paperless/src/documents/management/commands/document_importer.py", line 86, in handle
    self._import_files_from_manifest(options["no_progress_bar"])
  File "/usr/src/paperless/src/documents/management/commands/document_importer.py", line 165, in _import_files_from_manifest
    create_source_path_directory(document.archive_path)
  File "/usr/src/paperless/src/documents/file_handling.py", line 20, in create_source_path_directory
    os.makedirs(os.path.dirname(source_path), exist_ok=True)
  File "/usr/local/lib/python3.9/posixpath.py", line 152, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

Looks like a path is empty somewhere but I can't figure out where to look for this issue or how to solve it. At this stage Paperless-ngx shows the metadata for all documents but most of the pdfs are missing.

I'm out of ideas. It should be a simple migration but it seems to fail no matter how I try to do it. Migrating manually is not an option as there are way over 1k documents in paperless-ng.

I restored the backup again and so paperless-ng is running as before. No errors in the log, nothing looking wrong here. There should be a `document_sanity_checker` but I could not find a way to execute it. But this is 

**Relevant information**
 - Host OS of the machine running paperless: Debian Bullseye, Docker
 - Browser: Safari
 - Version (migration from): jonaswinkler/paperless-ng:latest
 - Version (migration to): ghcr.io/paperless-ngx/paperless-ngx:latest
 - Installation method: docker

docker-compose.env:
```
PAPERLESS_TIME_ZONE=Europe/Zurich
PAPERLESS_OCR_LANGUAGES=deu eng
PAPERLESS_FILENAME_FORMAT={created_year}/{correspondent}/{title}
PAPERLESS_IGNORE_DATES=19xx-yy-zz
PAPERLESS_CONSUMER_RECURSIVE=true
PAPERLESS_CONSUMER_SUBDIRS_AS_TAGS=true
```

docker-compose.yaml:
```
version: "3.4"
services:
  broker:
    image: redis:6.0
    restart: unless-stopped
    networks:
      - backend

  webserver:
    image: ghcr.io/paperless-ngx/paperless-ngx:latest
    restart: unless-stopped
    depends_on:
      - broker
    networks:
      - revproxy
      - backend
      - email
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - data:/usr/src/paperless/data
      - media:/usr/src/paperless/media
      - /var/lib/paperless/export:/usr/src/paperless/export
      - /var/lib/paperless/consume:/usr/src/paperless/consume
    env_file: docker-compose.env
    environment:
      PAPERLESS_REDIS: redis://broker:6379

volumes:
  data:
  media:

networks:
  backend:
  revproxy:
    external:
      name: rp
  email:
    external:
      name: email
```

**Comment by stumpylog** [comment]
The document sanity checker should work the same as the exporter, something like `docker exec --tty webserver manage document_sanity_checker`

Could you run that on the original `paperless-ng` container?

**Comment by tht** [comment]
Thanks for the fast response. Knowing that the sanity_checker *should* work did help to solve the issue.

I finally realized that I was running an old version of paperless-ng. This is how I was able to fix it:

`docker-compose pull` to get the current *latest*.

Executing `docker-compose exec webserver python3 manage.py showmigrations` I was able to identify the missing migrations:
```
 [ ] 1012_fix_archive_files
 [ ] 1013_migrate_tag_colour
 [ ] 1014_auto_20210228_1614
 [ ] 1015_remove_null_characters
 [ ] 1016_auto_20210317_1351
paperless_mail
 [ ] 0008_auto_20210516_0940
```

`docker-compose exec webserver python3 manage.py migrate` failed at first, complaining about a missing file. I added an empty file at this path and now the migration did run just fine.

Now I was able to switch to paperless-ngx without an issue.

Thanks for looking into this!
